src/components/Editor/ClientCursor.ts

```diff
@@ -15,7 +16,7 @@ class ClientCursor {
     return new ClientCursor(id, color);
   }
 
-  updateCursor(cursorPos: number, cm: any) {
+  updateCursor(cm: any, cursorPos: number) {
```

Comment:
cursorPos is not a numeric type. `posFromIndex` returns a `codemirror.Postion` object.
And the `cm` parameter can be specified as `codemirror.Editor`.

All of this can be checked by installing `@types/codemirror` as devDependence.
https://www.npmjs.com/package/@types/codemirror

And you will be able to modify it anywhere you use codemirror.

Reply:
When I add a type as shown below, it works properly after adding type definitions(`npm install -save-dev @types/codemirror`).

```typescript
import { Position } from 'codemirror';
// ...
updateCursor(cm: any, cursorPos: Position) {
```